addons/folo/common/src/main/java/org/commonjava/indy/folo/data/FoloRecordCassandra.java

```diff
@@ -0,0 +1,433 @@
+
+    }
+
+    @Override
+    public boolean hasRecord(TrackingKey key) {
+        return hasSealedRecord(key) || hasInProgressRecord(key);
+    }
+
+    @Override
+    public boolean hasSealedRecord(TrackingKey key) {
+
+        BoundStatement bind = getTrackingRecordsByTrackingKey.bind(key);
+        ResultSet execute = session.execute(bind);
+        Row one = execute.one();
+        if(one != null) {
+            Boolean sealed = one.getBool("sealed");
+            if(sealed) {
+                return true;
+            } else {
+                return false;
+            }
+        }else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean hasInProgressRecord(TrackingKey key) {
+        BoundStatement bind = getTrackingRecordsByTrackingKey.bind(key);
+        ResultSet execute = session.execute(bind);
+        Row one = execute.one();
+        if(one != null) {
+            Boolean sealed = one.getBool("sealed");
+            if(!sealed) {
+                return true;
+            } else {
+                return false;
+            }
+        }else {
+            return false;
+        }
+    }
+
+    @Override
+    public TrackedContent get(TrackingKey key) {
+        List<DtxTrackingRecord> trackingRecords =  getDtxTrackingRecordsFromDb(key);
+        return transformDtxTrackingRecordToTrackingContent(key,trackingRecords);
+    }
+
+    @Override
+    public TrackedContent seal(TrackingKey trackingKey) {
+        List<DtxTrackingRecord> trackingRecords =  getDtxTrackingRecordsFromDb(trackingKey);
+        DtxTrackingRecord recordCheck = trackingRecords.get(0);
+        if(recordCheck.getState()) {
+            logger.debug( "Tracking record: {} already sealed! Returning sealed record.", trackingKey );
+            return transformDtxTrackingRecordToTrackingContent(trackingKey,trackingRecords);
+        }
+        logger.debug( "Sealing record for: {}", trackingKey );
+        for(DtxTrackingRecord record : trackingRecords) {
+            record.setState(true);
+            trackingMapper.save(record);
+        }
+        return transformDtxTrackingRecordToTrackingContent(trackingKey,trackingRecords);
+    }
+
+    @Override
+    public Set<TrackingKey> getInProgressTrackingKey() {
+        BoundStatement inProgress = getTrackingRecordBySealed.bind(false);
+        ResultSet getInProgressRecords = session.execute(inProgress);
+        List<Row> allInProgress = getInProgressRecords.all();
+        Iterator<Row> iterator = allInProgress.iterator();
+
+        Set<TrackingKey> trackingKeys = new HashSet<>();
+        while (iterator.hasNext()) {
+            Row next = iterator.next();
+            String tracking_key = next.getString("tracking_key");
+            trackingKeys.add(new TrackingKey(tracking_key));
+        }
+        return trackingKeys;
+    }
+
+    @Override
+    public Set<TrackingKey> getSealedTrackingKey() {
+        BoundStatement inProgress = getTrackingRecordBySealed.bind(true);
```

Comment:
this will become a full table scan too. what if use the other builds table. does it contains the sealed ids?

Reply:
Yes it is full table scan but because of tracking_key as partition key. records will be grouped. by tracking_key on. cassandra nodes and queries will be. with highest performance.
Other table. is just informational. table where we would register start/finish. times for builds.

Reply:
I've suggested collapsing these two methods, in a comment above.

Reply:
> "I've suggested collapsing these two methods, in a comment above."

yes, i agree , it will be done with next commit.